arbnode/resourcemanager/resource_management_test.go

```diff
@@ -37,42 +36,57 @@ total_rss 1016209408
 total_inactive_file %d
 total_active_file 321544192
 `, inactive)
-	if err != nil {
-		return err
+	return err
+}
+
+func makeCgroupsTestDir(cgroupDir string) cgroupsMemoryFiles {
+	return cgroupsMemoryFiles{
+		limitFile:  cgroupDir + "/memory.limit_in_bytes",
+		usageFile:  cgroupDir + "/memory.usage_in_bytes",
+		statsFile:  cgroupDir + "/memory.stat",
+		inactiveRe: regexp.MustCompile(`total_inactive_file (\d+)`),
 	}
-	return nil
 }
 
-func TestCgroupsv1MemoryLimit(t *testing.T) {
-	cgroupDir := t.TempDir()
-	c := newCgroupsV1MemoryLimitChecker(cgroupDir, 95)
-	_, err := c.isLimitExceeded()
-	if err == nil {
-		t.Error("Should fail open if can't read files")
+func TestCgroupsFailIfCantOpen(t *testing.T) {
```

Comment:
func TestCgroupsv1MemoryLimit(t *testing.T) {

  for _, tc := range []struct {
        desc string
        inactive int
        want bool
} {
    {  
       desc: "limit should be exceeded", 
      inactive: 51,
      want: true,

    },
    {  
       desc: "limit should not be exceeded", 
      inactive: 51,
      want: false,

    },
} {
   t.Run(tc.desc, func(t *testing.T) {
         dir := makeCgroupsTestDir(t.TempDir())
         if err := updateFakeCgroupFiles(c, 1000, 1000, tc.inactive); err != nil {
      		t.Fatal("Updating cgroup files: %v", err)
         }
         exceeded, err := c.isLimitExceeded()
         if err != nil {
            t.Fatalf("Checking if limit exceeded: %v", err)
         }
         if exceeded != tc.want {
            t.Errorf("isLimitExceeded() = %t, want %t", exceeded, tc.want)
         }
	}
   })
}
}



You can either also have TestCgroupsFailIfCantOpen as another case in the test, or leave as is.

Reply:
You can see examples of table-driven tests at: https://github.com/OffchainLabs/nitro/blob/ac0a2de65e86b008ff10ded1f7b003ca01737315/arbnode/dataposter/storage_test.go

Reply:
Thanks, I'll use this pattern more often. Took your code largely as suggested.